odoo/sql_db.py

```diff
@@ -466,57 +423,20 @@ def _close(self, leak=False):
             self._cnx.leaked = True
         else:
             chosen_template = tools.config['db_template']
-            templates_list = tuple(set(['template0', 'template1', 'postgres', chosen_template]))
-            keep_in_pool = self.dbname not in templates_list
+            keep_in_pool = self.dbname not in ('template0', 'template1', 'postgres', chosen_template)
             self.__pool.give_back(self._cnx, keep_in_pool=keep_in_pool)
 
     def autocommit(self, on):
+        warnings.warn(
+            f"Deprecated Methods since 16.0, use {'`_cnx.autocommit = True`' if on else '`_cnx.set_isolation_level`'} instead.",
```

Comment:
Shouldn't this forbid setting autocommit as it's been deprecated since 13.0 because it's broken?

Reply:
I will add extra complexity to forbid it then no. I prefer like this, and we will remove all the method soon anyway.